src/Pos/Util/BackupPhrase.hs

```diff
@@ -31,8 +31,8 @@ newtype BackupPhrase = BackupPhrase
     } deriving (Eq, Generic)
 
 instance Bi BackupPhrase where
-    sizeNPut = labelS "BackupPhrase" $ putField bpToList
-    get = label "BackupPhrase" $ BackupPhrase <$> get
+  encode = encode
```

Comment:
Weird, it should have been `encode .  bpToList`, it seems. I suspect what happened here (alas!) is that `BackupPhrase` derives `Generic`, so `encode` here silently typechecked and sneakily used the generic implementation of decode. Worst of all, I couldn't test this in my specs as apparently `BackupPhrase` doesn't have an `Arbitrary` instance defined. Awesome catch, @gromakovsky !